jenkins-plugin/src/main/java/org/jenkinsci/plugins/pipeline/maven/WithMavenStepExecution.java

```diff
@@ -715,42 +772,79 @@ private String setupGlobalSettingFile(@Nonnull Collection<Credentials> credentia
             FilePath settings;
             if ((settings = ws.child(settingsPath)).exists()) {
                 // Global settings file residing on the agent
-                console.format("[withMaven] use Maven global settings provided on the build agent '%s' %n", settingsPath);
-                LOGGER.log(Level.FINE, "Copying maven global settings file from build agent {0} to {1}", new Object[]{settings, settingsDest});
+                if (LOGGER.isLoggable(Level.FINE)) {
+                    console.format("[withMaven] using Maven global settings provided on the build agent '%s' %n", settingsPath);
+                    LOGGER.log(Level.FINE, "Copying maven global settings file from build agent {0} to {1}", new Object[] { settings, settingsDest });
+                }
                 settings.copyTo(settingsDest);
+                envOverride.put("GLOBAL_MVN_SETTINGS", settingsDest.getRemote());
+                return settingsDest.getRemote();
             } else {
                 throw new AbortException("Could not find file '" + settings + "' on the build agent");
             }
-            envOverride.put("GLOBAL_MVN_SETTINGS", settingsDest.getRemote());
-            return settingsDest.getRemote();
         }
 
         // Settings provided by the global maven configuration
-        GlobalSettingsProvider globalSettingsProvider = GlobalMavenConfig.get().getGlobalSettingsProvider();
+        GlobalSettingsProvider globalSettingsProvider;
+        MavenConfigFolderOverrideProperty overrideProperty = getMavenConfigOverrideProperty();
+
+        StringBuilder mavenSettingsLog = new StringBuilder();
+        if (overrideProperty != null && overrideProperty.getGlobalSettings() != null) {
+            // Settings overriden by a folder property
+            if (LOGGER.isLoggable(Level.FINE)) {
+                mavenSettingsLog.append("[withMaven] using overriden Maven global settings by folder '").append(overrideProperty.getOwner().getDisplayName()).append("'. ");
+            }
+            globalSettingsProvider = overrideProperty.getGlobalSettings();

```

Comment:
why not `overrideProperty.getSettings()` ? aren t we taking the specific settings defined on the folder here? (as u did above)

Reply:
but honestly I am not sure why you need this here, this is globalSettings of maven, can they be overridden?

Reply:
No. There are two different things. This is not the Global settings meaning Global jenkins instance, but the Maven Global settings. There are two types of settings files in Maven, what we call Maven Settings and Maven Global Settings, which can have different sets of configurations.